include/clap/ext/draft/undo-redo.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../string-sizes.h"
+#include "../../stream.h"
+
+/// @page Undo/Redo
+///
+/// For each action that causes a change of parameter values and non-parameter state
+/// (the 'plugin state'), the plugin provides an incremental change event object to the host.
+/// These incremental change event objects then can be applied in reverse order in order to
+/// undo those actions. If undone, the same objects can be used to redo the action again.

```

Comment:
Yes, my first assumption was that undo/redo objects could be the same / contain information for both directions. Thought a bit about it and that probably is not the best idea as it would mean a lot of redundant information as each object would have to store the value before and after the change. This would be especially 'nice' for preset changes where the 'value' actually might be the whole plugin state. 😆

Reply:
Hope I could resolve that first misunderstanding which the current iteration.